gene_transformer/blast.py

```
@@ -3,94 +3,87 @@
+        self.scores: List[float] = []
+
+    def _warning_message(self) -> None:
+        print(
+            "Scores have not yet been defined. Make sure"
+            " you've called run_blast() and get_scores()."
+        )
 
-    def run_blast(self):
+    def run_blast(self) -> None:
         # run local blastn given parameters in init, REQUIRES LOCAL INSTALLATION OF BLAST
         command = "blastn -query {} -subject {} -out {} -outfmt 10".format(
             self.temp_fasta, self.database_file, self.temp_csv
         )
         subprocess.run(command, shell=True)
         self.ran_blast = True
 
-    def get_scores(self):
+    def get_scores(self) -> Optional[List[float]]:
+        if not self.ran_blast:
+            print("Blast has not yet been run. Call the run_blast() method first.")
+            return None
         try:
-            assert self.ran_blast
-            # read in csv where blast results were stored
+            # Read in csv where blast results were stored
             df = pd.read_csv(self.temp_csv, header=None)
-            # take column which specifies scores
+            # Take column which specifies scores
             self.scores = df[11].tolist()
             return self.scores
-        except AssertionError:
-            print("Blast has not yet been run. Call the run_blast() method first.")
-            return None
         except pd.errors.EmptyDataError:
             self.scores = [-1]
+            return None
 
-    def get_top_score(self):
-        try:
-            assert self.scores
+    def get_top_score(self) -> Optional[float]:
+        if self.scores:
             return self.scores[0]
-        except AssertionError:
-            print(
-                "Scores have not yet been defined. Make sure you've called run_blast() and get_scores()."
-            )
-            return None
 
-    def get_mean_score(self):
-        try:
-            assert self.scores
+        self._warning_message()
```

Should we have it return None with a warning? It might be better to have it raise an Exception if this is called before `run_blast()` since this means the code is not being used properly. Otherwise we might get silent errors. @maxzvyagin

---

Instead, we can simply run blast and get scores if they have not been run.